### PR TITLE
Upgrade zendframework/zend-diactoros to 2.x

### PR DESCRIPTION
**Pull request by MidnightDesign**

### CODE REVIEW

**Comment by MidnightDesign** [comment]
Nudge @reinink

**Comment by reinink** [comment]
Hey there! Have you actually tried this with the current version of Glide? Does it all work good? Meaning, can I merge this in without manually testing it with Glide?

**Comment by rieschl** [comment]
@reinink As far as I see, the `thephpleague/glide` just type hints to the PSR-7 interfaces, no actual implementation (as is diactoros) is used there. So there should be no implications.
But maybe it would be a better option to allow `zend-diactoros` versions `^1.0 || ^2.0`? That would probably make live easier for the consumers of the package.

**Comment by reinink** [comment]
@rieschl Yes, seems wise to me!

@MidnightDesign want to update?

**Comment by MidnightDesign** [comment]
Great idea. I've updated the PR.

File: tests/Responses/ZendResponseFactoryTest.php [hunk 1]
```diff
@@ -27,7 +28,10 @@ public function testCreate()
         $this->assertInstanceOf('Zend\Diactoros\Response', $response);
         $this->assertEquals('image/jpeg', $response->getHeaderLine('Content-Type'));
         $this->assertEquals('5175', $response->getHeaderLine('Content-Length'));
-        $this->assertContains(gmdate('D, d M Y H:i', strtotime('+1 years')), $response->getHeaderLine('Expires'));
```
Comment (reinink) [inline tests/Responses/ZendResponseFactoryTest.php]: Curious why you changed this from `gmdate` to `date_create`?

Comment (MidnightDesign) [reply tests/Responses/ZendResponseFactoryTest.php]: It's been too long, I can't remember. Should have specified that better. But the tests are working wothout the change, so I'm going to revert that.

**Comment by rieschl** [comment]
ping @reinink does anything prevent merging this PR? I can't update to `^2.0` in my project because of that constraint.

Thanks :)

**Comment by reinink** [comment]
What should I tag this release? 2? Seems like a breaking change...

**Comment by rieschl** [comment]
No, it's not a BC break. Consumers can still use `zendframework/zend-diactoros` and don't have to update anything in their code. I would tag a minor release `1.1.0`.
Or did I miss something @MidnightDesign ?

**Comment by MidnightDesign** [comment]
I don't see any BC breaks. Users with older PHP versions will just not get the new version, but it shouldn't break anything...

**Comment by reinink** [comment]
Merged and tagged 👍

**Pull request merged**